CRM/Contribute/Selector/Search.php

```diff
@@ -402,6 +402,13 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
         }
       }
 
+      $row['balance_due'] = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(
+        $result->contribution_id,
```

Comment:
On further research, if we MUST add this improvement i.e. add ```Balance Due``` column on Contribution list, then there is no other easy alternative other than using ```CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(..)``` to calculate the balance due amount.